contracts/factory/src/tests/queries.rs

```diff
@@ -390,3 +393,283 @@ fn test_queries_by_tuple_errors() {
+                            address: token1.address.clone(),
+                            amount: 150i128,
+                        },
+                        Asset {
+                            address: token2.address.clone(),
+                            amount: 200i128
+                        }
+                    )
+                }
+            ],
+            stake_portfolio: vec![&env,]
+        }
+    );
+
+    first_stake_client.bond(&user_1, &173i128);
+
+    // first portfolio after staking
+    let first_portfolio = factory.query_user_portfolio(&user_1, &true);
+    assert_eq!(
+        first_portfolio,
+        UserPortfolio {
+            lp_portfolio: vec![&env,],
+            stake_portfolio: vec![
+                &env,
+                StakePortfolio {
+                    staking_contract: first_stake_address.clone(),
+                    stakes: vec![
+                        &env,
+                        Stake {
+                            stake: 173i128,
+                            stake_timestamp: 0
+                        }
+                    ]
+                }
+            ]
+        }
+    );
+
+    let second_lp_init_info =
+        generate_lp_init_info(&token3, &token4, manager.clone(), &admin, fee_recipient);
+
+    let second_lp_contract_addr = factory.create_liquidity_pool(
+        &admin.clone(),
+        &second_lp_init_info,
+        &String::from_str(&env, "Second Pool"),
+        &String::from_str(&env, "PHO/ETH"),
+    );
+
+    let second_lp_client = lp_contract::Client::new(&env, &second_lp_contract_addr);
+    let second_stake_address = factory
+        .query_pool_details(&second_lp_contract_addr)
+        .pool_response
+        .stake_address;
+
+    let second_stake_client = stake_contract::Client::new(&env, &second_stake_address);
+
+    second_lp_client.provide_liquidity(
+        &user_2.clone(),
+        &Some(200),
+        &Some(100i128),
+        &Some(250),
+        &Some(100i128),
+        &None::<i64>,
+    );
+
+    let second_portfolio = factory.query_user_portfolio(&user_2, &true);
+    assert_eq!(
+        second_portfolio,
+        UserPortfolio {
+            lp_portfolio: vec![
+                &env,
+                LpPortfolio {
+                    assets: (
+                        Asset {
+                            address: token3.address.clone(),
+                            amount: 200i128,
+                        },
+                        Asset {
+                            address: token4.address.clone(),
+                            amount: 250i128
+                        }
+                    )
+                }
+            ],
+            stake_portfolio: vec![&env,]
+        }
+    );
+
+    second_stake_client.bond(&user_2, &223i128);
+
+    let second_portfolio = factory.query_user_portfolio(&user_2, &true);
+    assert_eq!(
+        second_portfolio,
+        UserPortfolio {
+            lp_portfolio: vec![&env,],
```

Comment:
Why in this case is `lp_portfolio` an empty vector?

Reply:
we were querying for lp_share tokens only in the pool and not the ones that were sent by the user to the corresponding staking contract. Thanks